### DISCUSSION TITLE
Data Validation in Prisma

### DISCUSSION THREAD

**Discussion by MohamadrezaKazemian** [prisma/prisma#17009] [Q&A] [answered]
been working with Prisma lately and recently I needed to validate the data coming from the client.
for instance, in express-js, there is something called EXPRESS-VALIDATOR to validate the data.
do we have anything like that in Prisma? or should we do it manually like checking the length. . .
thanks for your help.

**Comment by pantharshit00** [comment, marked as answer]
You can use a Prisma client extension here. Checkout this example on how you can achieve that: https://github.com/sbking/prisma-client-extensions/tree/main/examples/input-validation

**Reply by MohamadrezaKazemian** [reply]
> You can use a Prisma client extension here. Checkout this example on how you can achieve that: https://github.com/sbking/prisma-client-extensions/tree/main/examples/input-validation

So I should use another package called ''Z" to approach validation right?

**Reply by MohamadrezaKazemian** [reply]
is there any more straightforward way to do that? for example, just checking if the input is email or is it between 5 to 10 characters.

**Reply by pantharshit00** [reply]
As mentioned by @IconicDexter, Zod provides you with methods in which you can express this validation.

**Comment by knnedy** [comment]
You can use a package called Zod its great for TypeScript because of type safety or use Joi(Hapi Joi).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
